src/plugins/intel_cpu/src/nodes/transpose.cpp

```diff
@@ -119,7 +119,16 @@ void Transpose::initSupportedPrimitiveDescriptors() {
     } else {
         // general plain case
         config.inConfs[0].setMemDesc(creatorsMap.at(LayoutType::ncsp)->createSharedDesc(prec, inputDataShape));
-        config.outConfs[0].setMemDesc(creatorsMap.at(LayoutType::ncsp)->createSharedDesc(prec, outputDataShape));
+        std::shared_ptr<CpuBlockedMemoryDesc> outMemDescPtr;
+        if (isOptimized) {
```

Comment:
This still looks a bit hacky, shouldn't we instead fuse transpose into FC node, basically extending semantics of the node to support different layouts / shapes on the weights input?
@dmitry-gorokhov @antonvor